powdr_cli/src/main.rs

```diff
@@ -180,22 +215,18 @@ fn main() {
             output_directory,
             inputs,
             force,
-        } => match field {
-            FieldArgument::Gl => compiler::compile_pil_or_asm::<GoldilocksField>(
-                &file,
-                split_inputs(&inputs),
-                Path::new(&output_directory),
-                force,
-            ),
-            FieldArgument::Bn254 => compiler::compile_pil_or_asm::<Bn254Field>(
-                &file,
-                split_inputs(&inputs),
-                Path::new(&output_directory),
-                force,
-            ),
-        },
+            prove,
+        } => call_with_field!(
```

Comment:
Maybe we should use a settings struct instead of a macro? Also not sure if that would help.

Reply:
I'm not sure a macro is the best way to do it, but at some point we need to go from an enum variant to a type parameter to call the function with. Would the settings struct have the enum in it? Wouldn't that just move the problem to the next step?

Reply:
Yes, it wolud only move it to the next step, but my point is that when we have a settings struct, we only need to provide one parameter:
```
if fiel is bla {
  func<BlaField>(my_struct);
} else {
  funct<OtherField>(my_struct);
}
```

Reply:
Ahhhh you're talking about the arguments, not the generics, I see. I'll do that in a separate PR.